packages/p2p-v2/src/pages/my-ads/screens/MyAds/MyAdsTable/MyAdsTable.tsx

```diff
@@ -49,27 +51,54 @@ const MyAdsTable = () => {
     const { data: advertiserInfo } = p2p.advertiser.useGetInfo();
     const { mutate } = p2p.advert.useUpdate();
     const { mutate: updateAds } = p2p.advertiser.useUpdate();
+    const { error, isSuccess, mutate: deleteAd } = p2p.advert.useDelete();
     const { isDesktop } = useDevice();
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const [advertId, setAdvertId] = useState('');
+
+    useEffect(() => {
+        if (isSuccess) {
+            setAdvertId('');
+        }
+        if (error?.error?.message) {
+            setIsModalOpen(true);
+        }
+    }, [error?.error?.message, isSuccess]);
 
     if (isLoading) return <Loader />;
 
     const onClickIcon = (id: string, action: string) => {
-        //TODO: to implement the onclick actions
+        //TODO: to implement the onclick actions for share and edit.
         switch (action) {
-            case 'activate':
+            case AD_ACTION.ACTIVATE:
                 mutate({ id, is_active: 1 });
                 break;
-            case 'deactivate':
+            case AD_ACTION.DEACTIVATE:
                 mutate({ id, is_active: 0 });
                 break;
-            case 'edit':
+            case AD_ACTION.DELETE: {
+                setAdvertId(id);
+                setIsModalOpen(true);
+                break;
+            }
             default:
                 break;
         }
     };
 
     const onClickToggle = () => updateAds({ is_listed: advertiserInfo?.is_listed ? 0 : 1 });
 
+    const onRequestClose = () => {
+        if (isModalOpen) {
+            setIsModalOpen(false);
+        }
+    };
+
+    const onClickDelete = () => {
+        deleteAd({ id: advertId });
+        onRequestClose();
```

Comment:
Just wondering how does this handle negative cases where deleteAd is not going through?

Reply:
for the error case, the error will be shown on the modal and for the other ones onrequestclose is being passed as props, so it closes the modal in those cases.